json11.cpp

```diff
@@ -364,13 +365,74 @@ struct JsonParser {
             i++;
     }
 
+    /* consume_comment()
+     *
+     * Advance comments (c-style inline and multiline).
+     */
+    bool consume_comment() {
+      bool comment_found = false;
+      if (str[i] == '/') {
+        i++;
+        if (i == str.size())
+          return fail("unexpected end of input inside comment", 0);
+        if (str[i] == '/') { // inline comment
+          i++;
+          if (i == str.size())
+            return fail("unexpected end of input inside inline comment", 0);
+          // advance until next line
+          while (str[i] != '\n') {
+            i++;
+            if (i == str.size())
+              return fail("unexpected end of input inside inline comment", 0);
+          }
+          comment_found = true;
+        }
+        else if (str[i] == '*') { // multiline comment
+          i++;
+          if (i == str.size())
+            return fail("unexpected end of input inside multi-line comment", 0);
+           // advance until closing tokens
+          while (!(str[i] == '*' && str[i+1] == '/')) {
```

Comment:
str[i+1] here is still an unchecked value.  Should check against size-1 above, and below, since it takes at least 2 characters to terminate a multi-line comment.